### DISCUSSION TITLE
Using buildah as a golang module

### DISCUSSION THREAD

**Discussion by abdelghanimeliani** [containers/buildah#4759] [General]
Hello i hope you are dowing well.
I am trying to use buildah in some project inspired from [this tutorial](https://github.com/containers/buildah/blob/main/docs/tutorials/04-include-in-your-build-tool.md) , the goal is to build an image from an existing image checkpoint , [the code](https://github.com/abdelghanimeliani/image-builder) works fine and the image was built.
![image](https://user-images.githubusercontent.com/62490938/234272445-cdab19fa-ace7-4353-acf7-017052a38a89.png)

i tried then to build some kubernetes operator with the operator framework with the same build logique used in  my previous code and the tutorial ,  and i tried to run it localy and i got this error 

> checking on destination /var/lib/containers/storage/overlay/1ac5f953c23fda28c389092fdf19ddaf42740d247efb2e5734e501dca9a9f554/merged: error in copier subprocess

when i added some fmt.print() to help me to debug i notice that the panic is happen [here](https://github.com/abdelghanimeliani/image-builder/blob/master/main.go#L76)  in the add function .

i tried then to add an api to the code in the endpoint /build using the gin package, i runned it in my localhost and i got this 
![image](https://user-images.githubusercontent.com/62490938/234271765-c779a540-db59-4135-bf2f-bef0f470c004.png)

so the question is , why do you think the code works fine when i started from the main methode and doens't works in other cases.

**Comment by rhatdan** [comment]
@giuseppe @nalind @flouthoc PTAL

**Comment by nalind** [comment]
When a subprocess is involved, the first thing I'd always check is that it's using `buildah.InitReexec()` correctly.  Is it being called from `main()` before any other output is produced?

**Reply by abdelghanimeliani** [reply]
@nalind no the init was in the handler methode , i moved it now in the top of the main() and that works , many thanks.

**Reply by flouthoc** [reply]
@abdelghanimeliani Could you please mark above thread as answer i'm closing this discussion
